tracer/src/Datadog.Trace/IAST/EvidenceConverter.cs

```diff
@@ -0,0 +1,240 @@
+// <copyright file="EvidenceConverter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Iast.SensitiveData;
+using Datadog.Trace.Iast.Settings;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Utilities;
+
+namespace Datadog.Trace.Iast;
+
+internal class EvidenceConverter : JsonConverter<Evidence>
+{
+    private bool _redactionEnabled;
+
+    public EvidenceConverter(bool redactionEnabled)
+    {
+        _redactionEnabled = redactionEnabled;
+    }
+
+    public override bool CanRead => false;
+
+    public override Evidence ReadJson(JsonReader reader, Type objectType, Evidence existingValue, bool hasExistingValue, JsonSerializer serializer)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void WriteJson(JsonWriter writer, Evidence evidence, JsonSerializer serializer)
+    {
+        if (_redactionEnabled)
+        {
+            WriteRedactedEvidence(writer, evidence, serializer);
+        }
+        else
+        {
+            WriteEvidence(writer, evidence);
+        }
+    }
+
+    private void WriteEvidence(JsonWriter writer, Evidence evidence)
+    {
+        writer.WriteStartObject();
+        if (evidence.Ranges == null || evidence.Ranges.Length == 0)
+        {
+            writer.WritePropertyName("value");
+            writer.WriteValue(evidence.Value);
+        }
+        else
+        {
+            writer.WritePropertyName("valueParts");
+            ToJsonTaintedValue(writer, evidence.Value!, evidence.Ranges);
+        }
+
+        writer.WriteEndObject();
+    }
+
+    private void ToJsonTaintedValue(JsonWriter writer, string value, Range[] ranges)
```

Comment:
From this method onwards this file is only 2 characters different from `SourceConvert.cs`, so I think we should add a common module for these methods.